File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.go
func maxMatrixSum(matrix [][]int) int64 {
    n := len(matrix)
    maximumNeg := math.MaxInt
    ans := int64(0)
    cnt := 0

    for i := 0; i < n; i++ {
        for j := 0; j < n; j++ {
            if matrix[i][j] < 0 {
                cnt++
            }
            num := int(math.Abs(float64(matrix[i][j])))
            maximumNeg = min(maximumNeg, num);
            ans += int64(num)
        }
    }

    if cnt % 2 == 1 {
        ans -= 2 * int64(maximumNeg)
    }
    return ans
}

File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.cpp
using ll = long long;
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        int n = matrix.size();
        int maximumNeg = INT_MAX;
        ll ans = 0;
        int cnt = 0;
        bool existZero = false;

        for(int i = 0; i < n; i++) {
            for(int j = 0; j < n; j++) {
                if(matrix[i][j] < 0) {
                    cnt++;
                } else if(matrix[i][j] == 0) {
                    existZero = true;
                }
                maximumNeg = min(maximumNeg, abs(matrix[i][j]));
                ans += abs(matrix[i][j]);
            }
        }

        if(cnt % 2 && !existZero) ans -= 2 * maximumNeg;
        return ans;
    }
};